packages/lib/src/interactions/RatioSelectionRect.tsx

```diff
@@ -16,9 +16,13 @@ function RatioSelectionRect(props: Props) {
 
   const { dataToWorld, worldToData, visSize } = useAxisSystemContext();
 
-  const ratioEndPoint = getRatioEndPoint(startPoint, endPoint, ratio);
+  const [ratioStartPoint, ratioEndPoint] = getRatioRespectingRectangle(
+    startPoint,
+    endPoint,
+    ratio
+  );
```

Comment:
If the center stays at the same place, it means the `startPoint` must move. This means that the code I wrote by assuming `endPoint` is the only vector to move must be refactored.

Therefore, I change the function returning the modified `endPoint` by a function that can modify `startPoint` and `endPoint`.